test/framework/clustervalidator.go

```diff
@@ -88,19 +93,50 @@ func NewClusterValidator(opts ...ClusterValidatorOpts) *ClusterValidator {
 	return &cv
 }
 
-type clusterOpts struct {
+// ClusterValidatorConfig holds the data required to perform validations on a cluster.
+type ClusterValidatorConfig struct {
 	ClusterClient           client.Client // the client for the cluster
 	ManagementClusterClient client.Client // the client for the management cluster
 	ClusterSpec             *cluster.Spec // the cluster spec
 }
 
-func validateEKSAObjects(ctx context.Context, validateOpts clusterOpts) error {
-	clusterClient := validateOpts.ClusterClient
-	if validateOpts.ClusterSpec.Cluster.IsManaged() {
-		clusterClient = validateOpts.ManagementClusterClient
+// GetWorkerNodeMachineSets gets a list of MachineSets corresponding the provided WorkerNodeGroupConfiguration from the management cluster.
+func (c *ClusterValidatorConfig) GetWorkerNodeMachineSets(ctx context.Context, w v1alpha1.WorkerNodeGroupConfiguration) ([]v1beta1.MachineSet, error) {
+	md := &v1beta1.MachineDeployment{}
+	mgmtClusterClient := c.GetManagementClusterClient()
+	mdName := fmt.Sprintf("%v-%v", c.ClusterSpec.Cluster.Name, w.Name)
+	key := types.NamespacedName{Name: mdName, Namespace: constants.EksaSystemNamespace}
+	if err := mgmtClusterClient.Get(ctx, key, md); err != nil {
+		return nil, fmt.Errorf("failed to get machine deployment %s when validating worker node: %v", w.Name, err)
+	}
+	ms := &v1beta1.MachineSetList{}
+	err := mgmtClusterClient.List(ctx, ms, client.InNamespace(constants.EksaSystemNamespace), client.MatchingLabels{
+		"cluster.x-k8s.io/deployment-name": md.Name,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get machine sets for deployment %s: %v", md.Name, err)
+	}
+	if len(ms.Items) == 0 {
+		return nil, fmt.Errorf("invalid number of machine sets associated with worker node configuration %v", w.Name)
+	}
+	return ms.Items, nil
+}
+
+// GetManagementClusterClient returns the client for the management cluster.
+// If the cluster is self-managed, it returns the clusters own client.
+func (c *ClusterValidatorConfig) GetManagementClusterClient() client.Client {
+	if c.ClusterSpec.Cluster.IsManaged() {
+		return c.ManagementClusterClient
 	}
+	return c.ClusterClient
+}
 
-	for _, obj := range validateOpts.ClusterSpec.ChildObjects() {
+// TODO: Move Validations to separate package
```

Comment:
Since we are already altering all of the validations in this PR, can we go ahead & move it to the other package in this PR? Or will that involve a significant amount of additional changes?

Reply:
My thoughts were to move the validations and the cluster validator out of the test folder as well. If we do that, then unit tests would be required for all of the validations, so I'm thinking it would be significant.

Reply:
Ok interesting! So since you're thinking to move it outside of the test folder, are we thinking to use the cluster validator for more than just e2e tests?

Reply:
Yeah, that's what I'm thinking

Reply:
Cool, thanks!